docs/conf.py

```diff
@@ -2,11 +2,27 @@
 
 import os
 import sys
+from sphinx_gallery.sorting import ExplicitOrder
+import wgpu.gui.offscreen
 
 
 ROOT_DIR = os.path.abspath(os.path.join(__file__, "..", ".."))
 sys.path.insert(0, ROOT_DIR)
 
+# -- Sphix Gallery Hackz -----------------------------------------------------
+# When building the gallery, render offscreen and don't process
+# the event loop while parsing the example
+
+
+def _ignore_offscreen_run():
+    wgpu.gui.offscreen.run = lambda: None
```

Comment:
After diving in (much) deeper as part of #362 I would say this is a safe choice when building the docs.

I see that in the scraper you call `canvas.draw()` which bypasses the event loop, which is why it works.

However, we should be mindful that if an example performs setup or initialization of the example in either the `animate`, `before_render` or `if __name__ == "__main__":` blocks, that code will not be executed (since it's pending on the event loop), and we'd need to refactor the example for it to appear correctly in the gallery.

No need to change anything. I just hope that makes sense and wanted to ensure we're all aware of this "gotcha".

Reply:
I'm not sure I follow. Doesn't the `animate` function get executed as part of `canvas.draw`? Otherwise, why does `plot_cube.py` show a rotating cube in the animation when I make repeated calls to `canvas.draw`?

Reply:
> Doesn't the animate function get executed as part of canvas.draw?

Yes it should. Unless I'm mistaken, Korijn's comment would only apply to code that explicitly schedules an asyncio task (well and the `if __name__ == "__main__"`).

Reply:
Inside sphinx-gallery, the `if __name__ == "__main__"` block will get executed. Examples are run as a "fake main" to replicate the behavior as-if it was run standalone.

Reply:
With https://github.com/pygfx/wgpu-py/pull/310, we should be all good (once we have a new wgpu-py release). I think we can resolve this and merge, right @Korijn ?